crypto/curve25519/curve25519.c

```diff
@@ -261,27 +263,192 @@ static void x25519_s2n_bignum_public_from_private(
 #endif
 }
 
-// Stub function until ED25519 lands in s2n-bignum
 static void ed25519_public_key_from_hashed_seed_s2n_bignum(
   uint8_t out_public_key[ED25519_PUBLIC_KEY_LEN],
   uint8_t az[SHA512_DIGEST_LENGTH]) {
+
+  uint64_t uint64_point[8] = {0};
+  uint64_t uint64_hashed_seed[4] = {0};
+  OPENSSL_memcpy(uint64_hashed_seed, az, 32);
+
+#if defined(OPENSSL_X86_64)
+
+  if (curve25519_s2n_bignum_no_alt_capable() == 1) {
+    edwards25519_scalarmulbase(uint64_point, uint64_hashed_seed);
+  } else if (curve25519_s2n_bignum_alt_capable() == 1) {
+    edwards25519_scalarmulbase_alt(uint64_point, uint64_hashed_seed);
+  } else {
+    abort();
+  }
+
+#elif defined(OPENSSL_AARCH64)
+
+  if (curve25519_s2n_bignum_alt_capable() == 1) {
```

Comment:
Does the order of these condition checks matter? For example for x86-64 we check `curve25519_s2n_bignum_no_alt_capable() == 1` first, but here for aarch64 we check `curve25519_s2n_bignum_alt_capable() == 1` first. Same question for similar code further below.

Reply:
Yes, definitely.

It's explained here: https://github.com/aws/aws-lc/blob/main/crypto/curve25519/curve25519.c#L164-L184. It uses the same reasoning so it wasn't changed for this PR.